parsigs/parse_sig_api.py

```diff
@@ -344,10 +358,43 @@ def _get_times_from_latin(frequency):
         return latin_freq.times
 
 
+def _get_interval_from_latin(frequency):  # yoad added this for cases like qxd/qxh (the interval changes and times is 1)
+    latin_freq = _get_latin_frequency(frequency)
+    if latin_freq:
+        return latin_freq.interval
+    else:
+        return 1
+
+
 def _get_latin_frequency(frequency):
     for latin_freq in latin_frequency_types.keys():
         if latin_freq in frequency:
             return latin_frequency_types[latin_freq]
+        # yoad added from here
+    for word in frequency.split(" "):
+        match_list = split_string_qx(word)
+        if is_latin_hour_abbreviation(match_list):
+            return _QXD_Frequency("Hour", 1, int(match_list[1]))  # the 1 index is the hour for example q4h is once every 4 hours
+        if is_latin_day_abbreviation(match_list):
+            return _QXD_Frequency("Day", 1,int(match_list[1]))  # the 1 index is the day for example q4d is once every 4 days
+
+
+def split_string_qx(string):
+    """Splits a string of the format 'q[number][h|d]' into three parts."""
+    string = string.replace('.', '')
+    match = re.match(r'([qQ])(\d+)([hHdD])', string)  # q or Q followed by number followed by either (h,H,d,D) hour/day
+    if match:
+        return list(match.groups())
+    else:
+        return None  # Or handle invalid strings as needed
+
+
+def is_latin_hour_abbreviation(frequency):
```

Comment:
This code tries to be generic and solve the latin case, but since there arent many permutation I suggest simply add them all to the existent dictionary